tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php

```diff
@@ -86,6 +86,25 @@ public function dataGetPlaceholderPositions()
             ['SELECT data.age AS age, data.id AS id, data.name AS name, data.id AS id FROM test_data data WHERE (data.description LIKE :condition_0 ESCAPE "\\\\") AND (data.description LIKE :condition_1 ESCAPE \'\\\\\') ORDER BY id ASC', false, [121 => 'condition_0', 174 => 'condition_1']],
             ['SELECT data.age AS age, data.id AS id, data.name AS name, data.id AS id FROM test_data data WHERE (data.description LIKE :condition_0 ESCAPE `\\\\`) AND (data.description LIKE :condition_1 ESCAPE `\\\\`) ORDER BY id ASC', false, [121 => 'condition_0', 174 => 'condition_1']],
             ['SELECT data.age AS age, data.id AS id, data.name AS name, data.id AS id FROM test_data data WHERE (data.description LIKE :condition_0 ESCAPE \'\\\\\') AND (data.description LIKE :condition_1 ESCAPE `\\\\`) ORDER BY id ASC', false, [121 => 'condition_0', 174 => 'condition_1']],
+            // Named parameter containing comment(s) with odd number of single quotes
+            [
+                <<<'SQLDATA'
+SELECT * FROM foo WHERE 
+bar=:a_param1
+-- Oops! Perfectly normal comment but it's got an odd number of apostrophes
+OR bar=':not_a_param1'
+OR bar=:a_param2
+OR bar=:a_param3
+OR bar=':not_a_param2'
+SQLDATA
+                ,
+                false,
+                [
+                    29 => 'a_param1',
+                    145 => 'a_param2',
+                    162 => 'a_param3',
+                ],
+            ],
 
```

Comment:
@TPHRyan could you add an extra data entry where a multi-line string literal would contain a line starting with `--`? E.g.:
```sql
INSERT INTO notes(text) values('
-- This is a comment
And this is the note itself
')
```

Reply:
Good point. It doesn't break with the current regex but I can see how it might.
Have pushed a further test for this case.